### ISSUE TITLE
Reading : BitFit Simple Parameter-efficient Fine-tuning for Transformer-based Masked Language-models

### ISSUE THREAD

**Issue by tikiti-kyotaro** [tikiti-kyotaro/Paperlog#11]
**Abstract**
We introduce BitFit, a sparse-finetuning method where only the bias-terms of the model (or a subset of them) are being modified. We show that with small-to-medium training data, applying BitFit on pre-trained BERT models is competitive with (and sometimes better than) fine-tuning the entire model. For larger data, the method is competitive with other sparse fine-tuning methods. Besides their practical utility, these findings are relevant for the question of understanding the commonly-used process of finetuning: they support the hypothesis that finetuning is mainly about exposing knowledge induced by language-modeling training, rather than learning new task-specific linguistic knowledge.

# 1. どんなもの？
Bit-fit を提案：モデルのバイアス項だけをチューニングする
小規模から中規模のデータではFTと同等もしくは上回る性能を誇る

FTはタスク特有の言語的知識を得るというよりも、トレーニング時に得た知識を公開する感じ？

BitFitの良さ
1. ごく少量のパラメータだけチューニング
2. 変更するパラメータ量はタスクに依存しない
3. 適応するパラメータは独立しており、局所的である（他のパラメータが連動して変わることがない？）
4. 小〜中量のデータであればFTと同程度かそれ以上の結果が出る
5. ちょっとの性能低下を良しとするならば、queryとmodel-of-NLPのバイアスという全体のバイアスの半分ほどのパラメータで十分

理想的なチューニングとは？
1. FTと同等な性能
2. パラメータの少しの部分だけの変更だけでいい
3. 全てのデータセットにアクセスする必要がない
4. タスク間で一貫性があること

チューニングの性質の分析
チューニングで新しい知識を学習する vs チューニングですでに知っていた知識を引き出せるようになる

# 2. 手法
BitFit (BIas-Term-FIne-Tuning) ：バイアスとタスクごとの分類器のみのチューニング（全体の0.1%）
3つのプロパティ
1. FTの性能と同等
2. 全てのデータにアクセスする必要はない
3. パラメータの一部分だけのチューニングで十分

# 3. 設定
データセット：GLUEベンチマーク
モデル：BERT_base, BERT-large, RoBERTa-BASE

# 4. 結果
1. AdapterとDiff-PruningとBitfitの比較(Table1)
		validationではBitFitはDiff-Pruneを4/9上回る→パラメータサイズは1/6のためすごい
		testではDiff-Pruneに2勝、Adapterに4勝(45倍のパラメータ)

2. 異なるモデルサイズでの比較(Table2)
		モデルサイズが大きくなっても結果は変わらない

3. バイアスは特別なパラメータなのか？(Table3)
		rand uniform がバイアスと同じパラメータをランダムにサンプルしてFTする
		バイアスをいじるBitFitの方が全然性能がいい

4. バイアスのサブセットをチューニングしてみる! (Figure1)
		bk : 変更しない
		bq : クエリのバイアスのみ
		bm2 : MLP層の内部バイアス
		Table3を見るとbqとbm2だけを変更した場合の結果が載っている→少ない劣化でパラメータ半分

5. 生成のギャップ
		訓練誤差とテストの誤差のギャップがBitFitだと少なくなっている

6. トークンレベルのタスク
		FTとほぼ同性能（詳しいのは載っている）

7. 訓練データのサイズ(Figure2)
		訓練データのサイズが小さいときはFTを上回る

# 5. まとめ
バイアスだけをチューニングするBitFitを提案
少量のパラメータ変更だけでタスク展開が容易
